content/waku/2024-gantt.md

```diff
@@ -0,0 +1,97 @@
+# Waku Roadmap 2024 Gantt Chart
+
+Status short term work only:
+
+- reliability for 1:1 chat and communities
+- up to 100 communities
+
+Colour legend:
+
+- **Red**: engineering work to deliver the feature.
+- Other: test and telemetry work to ensure quality
+
+golang engineers: working on status-go/go-waku
+nim engineers: working on nwaku, may be a CC from research team too.
+
+```mermaid
+gantt
+    dateFormat YYYY-MM-DD
+    axisFormat %d-%b
+    
+%% Milestones overview with deliverables
+    section Store Service Upgrade
+        Store v3-beta (msg hash): crit, milestone, after storev3-br, 0
+        Store v3 (sync): crit, milestone, after storev3-r storev3-g storev3-n, 0
+        DoS protection for req-res protocols: crit, milestone, after dosreqresn dosreqresj dosreqresg, 0
+        PostgreSQL maintenance: crit, milestone, after pgsql, 0
+        (metric) Count store messages: milestone, after count-store-msg, 0
+    section Direct Message Reliability
+        (testing) direct messages: milestone, after test-direct-msg, 0
+        Review connection management: crit, milestone, after review-conn-mgmt-1 review-conn-mgmt-2, 0
+        (telemetry) Fleet logging: milestone, after telem-fleet-logging, 0
+        (telemetry) direct message reliability: milestone, after telem-d-msg-rel, 0
+        Reliability Protocol for Relay: crit, milestone, after rel-relay-g rel-relay-n, 0
+        Reliability Protocol for Resource-Restricted Clients: crit, milestone, after rel-reqres-g rel-reqres-j, 0
+        Review MVDS usage and fail path: crit, milestone, after mvds, 0
+    section Static Sharding - dedicated shards
+        (telemetry) Measure Bandwidth: milestone, after telem-bandwidth, 0
+        Sharding peer mgmt and discovery hardening: crit, milestone, after sh-peer-mgmt-j sh-peer-mgmt-n sh-peer-mgmt-g, 0
+        (testing) Custom shard impl of Communities: milestone, after test-custom-shard, 0
+
+%% Tasks
+    section golang eng 1
+        %% Estimation TBC
+        Review connection management: crit, review-conn-mgmt-1, 2024-05-13, 8w
+        %% Estimate TBC
+        Review MVDS usage and fail path: crit, mvds, after review-conn-mgmt-1, 6w
+        %% Estimate/start TBC
+        (testing) direct messages: test-direct-msg, after mvds, 4w
+        %% Estimate TBC
+        (testing) Custom shard impl of Communities: test-custom-shard, after test-direct-msg, 4w 
+    section golang eng 2
+        Reliability Protocol for Relay (status-go): crit, rel-relay-g, 2024-05-13, 12w
+    section golang eng 3
+        Store v3 (go-waku client only): crit, storev3-g, 2024-02-26, 2024-05-24
+        %% Estimate TBC - assuming parallel work possible
+        Review connection management: crit, review-conn-mgmt-2, after storev3-g, 8w
+        %% TBC Estimate
+        Sharding peer mgmt and discovery hardening (go-waku): crit, sh-peer-mgmt-g, after review-conn-mgmt-2, 8w
```

Comment:
I think the 8w is realistic, given the many unknowns and parallel work here. This would include following any changes made in nwaku and ensuring resource-restricted clients can maintain the service node connectivity they need.

Reply:
Some form of shard aware peer management is already impl. in nwaku (behind a flag) and I did spend time testing it.

Should be easy to impl. in go